server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -648,8 +648,10 @@ public void lock(List<Pool> poolsToLock) {
             "FROM Pool AS Pool " +
             "  INNER JOIN Pool.product AS Product " +
             "  INNER JOIN Product.attributes AS Attribute " +
+            "  LEFT JOIN Pool.entitlements AS Entitlement " +
```

Comment:
With large amount of entitlements, I think this might be very slow. What do you think about using EXISTS clause instead of the left join?

What I mean is adding, into the where clause, an OR that will check whether Entitlement exists such that it entitles the Pool, and at the same time has endDateOverride >= current_date()